crates/bevy_render/src/render_resource/shader.rs

```diff
@@ -449,24 +456,34 @@ impl ShaderProcessor {
         shader_defs: &[ShaderDefVal],
         shaders: &HashMap<Handle<Shader>, Shader>,
         import_handles: &HashMap<ShaderImport, Handle<Shader>>,
+    ) -> Result<ProcessedShader, ProcessShaderError> {
+        let mut shader_defs_unique =
+            HashMap::<String, ShaderDefVal>::from_iter(shader_defs.iter().map(|v| match v {
+                ShaderDefVal::Bool(k, _) | ShaderDefVal::Int(k, _) | ShaderDefVal::UInt(k, _) => {
+                    (k.clone(), v.clone())
+                }
+            }));
+        self.process_inner(shader, &mut shader_defs_unique, shaders, import_handles)
+    }
+
+    fn process_inner(
```

Comment:
I'm not sure I understand why this needed to be split in process/process_inner

Reply:
Ah, it's because it's used in apply_imports

Reply:
Processing a shader need to be able to call itself recursively, and now to modify the shader defs while a given shader is processed. But values defined in one shader should not impact another, so the shader defs modifications are not bubbled up to the outside caller of `process`